### ISSUE TITLE
Using ```-Djdk.gtk.version=2``` to open jar file in Linux

### ISSUE THREAD

**Issue by Nephelite** [nus-cs2103-AY2223S1/forum#246]
I cannot run my jar file in Linux VM using only ```$ java -jar duke.jar```, nor double clicking. I have to use ```$ java -jar -Djdk.gtk.version=2 duke.jar``` in order to open it. From there, it runs properly as intended.

This is not a Linux VM only issue, my friend (not taking the module this semester) also had to run the latter command to get my jar file to work. He also could not double click to open the jar files.

Are incompatible gtk versions going to be considered as a bug? Do we have to fix this such that Linux users can simply run ```$ java -jar duke.jar``` or simply double click?

![image](https://user-images.githubusercontent.com/97427604/190856185-81599e7e-8e9d-45b7-88f7-8a2ae3fc27e4.png)
The exception if I run the former command. Screenshot because I cannot copy and paste from a VM.

**Comment by damithc** [comment]
@Nephelite Which OS did you use to create this JAR file?
Are you able to run JAR files created by other students (several were shared in other threads in this forum) in your Linux VM?

**Comment by Nephelite** [comment]
@damithc I used Windows to make the JAR file.

I can run JAR files by all the other students I've tested on if I use the ```-Djdk.gtk.version=2``` flag. In particular, I have tried it on at least 6 other students (semi-smoke testing for them, some not on the forum), and I could run the JAR file without the flag in only one of them. 

In other words, 5/6 of the JAR files I tried to open with ```$ java -jar duke.jar``` failed to open due to the exact exception in the image of the post. They all needed the flag to be run. Only 1 of the 6 could be double clicked or opened with ```$ java -jar duke.jar```.

And just to clarify, this is likely not an issue with my VM. I have a friend that also uses Linux, and he has the same issues. Not saying this issue is definitely a Linux issue though, it might be an issue only affecting a subset of Linux users.

**Comment by deepimpactmir** [comment]
You probably should bump your version of JavaFX to 11.0.2. Look at this for [more details](https://bugs.openjdk.org/browse/JDK-8210411). The issue is the version you're using does not support Wayland which is the default compositor on Ubuntu.

**Comment by parnikkapore** [comment]
It looks like JavaFX and Wayland don't get along well. (It's a new Linux GUI system that  is used by default in Ubuntu 22.04.) I can run my jar under an X11 session no problem, but not under a Wayland session.

Agreed about bumping JavaFX version.

**Comment by Nephelite** [comment]
@deepimpactmir @parnikkapore Thank you both so much for the input! 

Since this is an issue with newer Ubuntu systems, it should definitely be treated as a bug and addressed. I added the following to my ```build.gradle``` and now it works properly with just double clicking.

```java
javafx {
    version = "11.0.2"
    modules = [ 'javafx.controls', 'javafx.fxml' ]
}
```

I'm not actually sure if the module part is needed, but I referenced @njxue when adding this code (His iP was the one I was talking about that worked on double click in Ubuntu).

Thanks again for the advice!

Post-closure edit: Also, I had ```id 'org.openjfx.javafxplugin' version '0.0.7'``` in my plugins block in ```build.gradle```, but njxue uses ```id 'org.openjfx.javafxplugin' version '0.0.10'```. This is needed for the javafx block to work.

Post-closure edit 2: Apparently the javafx block MUST be above dependencies/below plugins.
